pkg/virt-handler/virtwrap/manager.go

```diff
@@ -325,10 +325,14 @@ func NewLibvirtDomainManager(connection Connection, recorder record.EventRecorde
 func (l *LibvirtDomainManager) SyncVM(vm *v1.VM) error {
 	var wantedSpec api.DomainSpec
 	mappingErrs := model.Copy(&wantedSpec, vm.Spec.Domain)
+
 	if len(mappingErrs) > 0 {
 		return errors.NewAggregate(mappingErrs)
 	}
-	dom, err := l.virConn.LookupDomainByName(vm.GetObjectMeta().GetName())
+	// Construct the domain name with a namespace prefix. E.g. namespace>name
+	domName := fmt.Sprintf("%s>%s", vm.GetObjectMeta().GetNamespace(), vm.GetObjectMeta().GetName())
```

Comment:
Could we extract this into an extra functions?

Something like a `VMNamespaceKeyFunc(obj VM) (string, error)` and a counter part `SplitVMNamespaceKey(string) (namespace, name, err)`? I think it is important enough to get it's own functions.